### PR TITLE
Add GitHub issue and PR templates.

### PR DESCRIPTION
**Pull request by kirkrodrigues**
Adapted from the log4j1-appenders [repo](https://github.com/y-scope/log4j1-appenders/tree/23342e0a6b004bcf9e19e3bc3796312a49048421/.github)

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/bug-report.yml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    description: "The release version number or development commit hash that has the bug."
+    placeholder: "Version number or commit hash"
+  validations:
+    required: true
+
+- type: textarea
+  attributes:
+    label: Environment
+    description: "The environment in which you're using logback-appenders."
+    placeholder: "OS version, Java version, etc."
```
Comment (LinZhihao-723) [inline .github/ISSUE_TEMPLATE/bug-report.yml]:
```suggestion
    placeholder: "Platform, OS version, Java version, etc."
```

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -0,0 +1,7 @@
+# Description
```
Comment (LinZhihao-723) [inline .github/PULL_REQUEST_TEMPLATE.md:1]:
```suggestion
# References
<!-- Any issues or pull requests relevant to this pull request -->

# Description
```
We also have `References` setup in other repos (i.e. python ffi)

**Review by LinZhihao-723 — CHANGES_REQUESTED** [review]
Added some comments with suggested changes

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 2 above)
Comment (kirkrodrigues) [reply .github/PULL_REQUEST_TEMPLATE.md:1]: Yeah, I removed it since most contributors are confused about how to use it.

Comment (LinZhihao-723) [reply .github/PULL_REQUEST_TEMPLATE.md:1]: uh ok, got it

**Review by LinZhihao-723 — APPROVED** [review]

**Pull request merged**